### PR TITLE
[#9439] Fix submission button not working caused by serializing FeedbackQuestionAttributes

### LINKED ISSUE #9439: Submit button on student sessions submission page not working
- **Environment**: 

* Local server at `* V7.0.0`, `master` branch at commit a018a980b8b31f08b167752a448c750347bbabdd

**Steps to reproduce**

1) Start a new submission for a new session as a student.

2) Fill out the questions.

3) Click on submit button

**Expected behaviour**

* Success message should be displayed

**Actual behaviour**

* Page does not show any message to the user

### PR DESCRIPTION
**Pull request by Adoby7**
Fixes #9439 

**Outline of Solution**

Gson cannot serialize composed abstract class `FeedbackQuestionDetails` inside `FeedbackQuestionAttribute` automatically. Therefore, an adaptor is necessary to serialize it manually.

#### Explanation
When we call `JsonUtil.toJson(obj)`, Gson will implicitly call the `toJson(obj, obj.getClass())`. Therefore, the following usage is correct:
```java
FeedbackResponseDetails frd = new FeedbackTextResponseDetails("response");
JsonUtil.toJson(frd);
```
because `frd.getClass()` will return a concrete class `FeedbackTextResponseDetails`, and thus Gson knows how to serialize it.

However, now we have the following structure:
``` java
class FeedbackResponseAttributes {
    FeedbackResponseDetails responseDetails;
    // other variables
}
```
where an abstract class `FeedbackResponseDetails` is composed in `FeedbackResponseAttributes`.
Then, the following code will not work, 
```java
FeedbackResponseAttributes fra = new FeedbackResponseAttributes("...");
JsonUtil.toJson(fra);
```
because `toJson(fra, FeedbackResponseAttributes.class)` will be executed. Unfortunately, Gson will not call `toJson(responseDetails, responseDetails.getClass())` when it serializes composed `FeedbackResponseDetails` without adaptor. As a result, only variables in the base class can be serialized, and this causes error

### CODE REVIEW

**Comment by xpdavid** [comment]
@RonakLakhotia Please help to check if the solution works.

**Comment by RonakLakhotia** [comment]
Yes, will do so

File: src/test/java/teammates/test/cases/datatransfer/FeedbackResponseAttributesTest.java [hunk 1]
```diff
@@ -92,6 +94,34 @@ public void testSetResponseDetails_shouldDoDeepCopy() {
 
     }
 
+    @Test
+    public void testGetJsonString_withQuestionDetails_shouldSerializeToConcreteQuestionDetailsClass() {
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/datatransfer/FeedbackResponseAttributesTest.java]: On a relevant note. The point is to test the new adaptor `TeammatesFeedbackResponseDetailsAdapter`. So I guess the test should reside in `JsonUtilsTest.java`?

**Comment by RonakLakhotia** [comment]
The button works fine now. 👍

**Pull request merged**